### PR TITLE
Avoid printing to the standard output in the tests

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: test/tstDetailsTreeConstruction.cpp [hunk 1]
```diff
@@ -18,16 +18,13 @@
...
-#include <algorithm>
 #include <array>
 #include <bitset>
 #include <functional>
 #include <limits>
 #include <sstream>
 #include <vector>
 
-#define BOOST_TEST_MODULE DetailsTreeConstruction
-
```
Comment (masterleinad) [inline test/tstDetailsTreeConstruction.cpp:30]: Does this change belong here?

**Review by masterleinad — COMMENTED** [review]
Apart from the one question, this looks good.

File: test/tstDetailsTreeConstruction.cpp (hunk 1 above)
Comment (dalg24) [reply test/tstDetailsTreeConstruction.cpp:30]: This is arguable.  I am editing that file so I figure I might as well fix it.
I can do it elsewhere if you want me to.

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Review by aprokop — COMMENTED** [review]
What are the advantages of using `BOOST_MESSAGE` over `std::cout`?

**Comment by Rombur** [comment]
You can decide at runtime if the message should be printed on the screen

**Review by Rombur — APPROVED** [review]

**Comment by dalg24** [comment]
Adding reference to #479

**Pull request merged**